src/value_generators/test/DocumentGeneratorTestCases.yml

```diff
@@ -189,6 +189,35 @@ Tests:
             "data" : "aaaaaaaaaaoEtN" }}
         ]}
 
+  - Name: Array Generator creates an array with a repeated element
+    GivenTemplate:
+      a:
+        {^Array: {of: {^RandomInt: {min: 1, max: 4}}, number: 2}}
+    ThenReturns:
+      - {a: [{ "$numberLong": "1" }, { "$numberLong": "1" }]}
```

Comment:
uhhh this test case doesn't make sense to me... how does it result in 2 ones deterministically?

Reply:
Huh, I missed this. Conceptually it could do this, but why should it do it? 
If you want to force the functionality you can wrap a ^Repeat around the ^RandomInt. It would work for this case and the next.

Reply:
In fact, that would make a better test case as it would force duplicate remove fro the distinct generator. Unless I'm missing something, in which case we should add a comment about why this has to produce a repeated entry.

Reply:
I don't think `DistinctArray` supports `Repeat` yet because `Repeat` isn't a `Generator` class. I can work on changing this, but it might take a bit longer.

Reply:
It has been resolved, I think!